### PR TITLE
[Functions] Update function with deletion task id

### PR DESCRIPTION
**Pull request by rokatyy**
Jira - https://iguazio.atlassian.net/browse/ML-6422

This task is a small part of  https://iguazio.atlassian.net/browse/ML-3432 during which function deletion api of v2 was implemented. In the new implementation, we've introduced a background task that handles function deletion and returns a task name to the user.

During the initial implementation, we overlooked how the UI would recognize that a function is being deleted. This task addresses that oversight by ensuring that function["status"]["deletion_task_id"] is set to deletion_task_id before the function is deleted. This way, the UI can accurately indicate that the function is in the process of being deleted.

### CODE REVIEW

File: server/api/api/utils.py [hunk 1]
```diff
@@ -1343,6 +1352,27 @@ async def _delete_function(
     )
 
 
+async def _update_functions_with_deletion_task_ids(
+    db_session, functions, project, background_task_name
+):
+    semaphore = asyncio.Semaphore(
```
Comment (liranbg) [inline server/api/api/utils.py:1369]: I think it worth adding a log line with how many functions being updated, project name

**Review by liranbg — COMMENTED** [review]
well done 🎖️ 
minor comment here

File: server/api/api/utils.py [hunk 2]
```diff
@@ -1314,6 +1318,19 @@ async def _delete_function(
...
+        functions_count=len(functions),
+        project=project,
+    )
+
+    # update functions with deletion task id
+    await _update_functions_with_deletion_task_ids(
+        db_session, functions, project, background_task_name
+    )
+
     if len(functions) > 0:
```
Comment (TomerShor) [inline server/api/api/utils.py]: can `len(functions)` be Zero?
If so we can skip everything, no?

**Review by TomerShor — COMMENTED** [review]
Nicely done!

File: server/api/api/utils.py (hunk 2 above)
Comment (rokatyy) [reply server/api/api/utils.py]: @TomerShor technically, yes, it's possible, because user can send a deletion request of the function which doesn't exist

Comment (rokatyy) [reply server/api/api/utils.py]: @TomerShor moved function deletion under the condition 👍

Comment (TomerShor) [reply server/api/api/utils.py]: Then let's move the `len` check right after getting the functions.
If there are not function - log that there is nothing to do and return

File: server/api/api/utils.py [hunk 3]
```diff
@@ -1318,20 +1318,19 @@ async def _delete_function(
         project,
         function_name,
     )
+    if len(functions) > 0:
+        logger.debug(
+            "Updating functions with deletion task id",
+            function_name=function_name,
+            functions_count=len(functions),
+            project=project,
+        )
```
Comment (TomerShor) [inline server/api/api/utils.py]:
```suggestion
    if len(functions) == 0:
        logger.debug(
            "No functions found, skipping deletion", function_name=function_name, project=project
        )
        return
        
    logger.debug(
        "Updating functions with deletion task id",
        function_name=function_name,
        functions_count=len(functions),
        project=project,
    )
    
    ...
```

**Review by TomerShor — CHANGES_REQUESTED** [review]

**Review by TomerShor — APPROVED** [review]

**Review by liranbg — APPROVED** [review]

**Pull request merged**